django_firebase_auth/views.py

```diff
@@ -45,8 +48,36 @@ class EmailNotVerified(AuthError):
     error_type = 'EMAIL_NOT_VERIFIED'
 
 
-def _get_firebase_email(headers: HttpHeaders):
-    jwt = headers.get(HEADER)
+def authenticate(request: HttpRequest):
+    try:
+        email = _verify_firebase_account(request.headers)
+    except AuthError as ex:
+        return JsonResponse(ex.make_response_body(), status=401)
+
+    user = _get_or_create_user(email)
+    if user is None:
+        return JsonResponse(UserNotRegistered.make_response_body(), status=401)
+
+    login(request=request, user=user, backend=AUTH_BACKEND)
+    return JsonResponse({"status": "ok"})
+
+
+def _get_or_create_user(email: str) -> Optional[AbstractBaseUser]:
+    UserModel = get_user_model()
+    try:
+        return UserModel.objects.get(email=email)
+    except UserModel.DoesNotExist:
+        if not CREATE_USER_IF_NOT_EXISTS:
+            return None
+
+    user = UserModel.objects.create_user(email=email, is_active=True)
+    user.set_unusable_password()
+    user.save()
+    return user
```

Comment:
default User model does not provide any `email_verified` flag so I'm not handling it here